Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+        /// whether the map contains an IndoorDefinition.
+        private func setIndoorDatasource() async throws {
+            try await map.floorManager?.load()
+            // If an indoor definition exists in the map, it gets loaded and sets the IndoorsDataSource to pull information
+            // from the definition, otherwise the IndoorsDataSource attempts to create itself using IPS table information.
+            if let indoorPositioningDefinition = map.indoorPositioningDefinition {
+                try await indoorPositioningDefinition.load()
+                indoorsLocationDataSource = IndoorsLocationDataSource(definition: indoorPositioningDefinition)
+            } else {
+                indoorsLocationDataSource = try await createIndoorLocationDataSourceFromTables(map: map)
+            }
+            // This ensures that the details of the inside of the building, like room layouts are displayed.
+            for featLayer in map.operationalLayers {
+                if featLayer.name == "Transitions" || featLayer.name == "Details" {
+                    featLayer.isVisible = true
+                }
+            }
+            // The indoorsLocationDataSource should always be there. Since the createIndoorLocationDataSourceFromTables returns
+            // an optional value it cannot be guaranteed. Best option if you get to this point without a datasource
+            // is to return (ideally an error would have been thrown before this point and the flow broken.)
+            guard let dataSource = indoorsLocationDataSource else { return }
+            locationDisplay.dataSource = dataSource
+            locationDisplay.autoPanMode = .compassNavigation
+            // Start the location display to zoom to the user's current location.
+            try await locationDisplay.dataSource.start()
+        }
+        
+        /// Creates an indoor location datasource from the maps tables if there is no indoors definition.
+        /// - Parameter map: The map which contains the tables from which the data source is constructed.
+        /// - Returns: Returns a configured IndoorsLocationDataSource created from the IPS position table.
+        private func createIndoorLocationDataSourceFromTables(map: Map) async throws -> IndoorsLocationDataSource? {
+            // Gets the positioning table from the map.
+            guard let positioningTable = map.tables.first(where: { $0.displayName == "IPS_Positioning" }) else { return nil }
+            // Creates and configures the query parameters.
+            let queryParameters = QueryParameters()
+            queryParameters.maxFeatures = 1
+            queryParameters.whereClause = "1 = 1"
+            // Queries positioning table to get the positioning ID.
+            let queryResult = try await positioningTable.queryFeatures(using: queryParameters)
+            guard let feature = queryResult.features().makeIterator().next() else { return nil }
```

Comment:
Here's another way of doing this.

```suggestion
            guard let feature = queryResult.features().first(where: { _ in true }) else { return nil }
```

Reply:
Thanks for the tip!